main/src/ca/uwaterloo/flix/api/lsp/provider/InlayHintProvider.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Nicola Dardanis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider
+
+import ca.uwaterloo.flix.api.lsp.Entity.LocalVar
+import ca.uwaterloo.flix.language.ast.TypedAst.Root
+import ca.uwaterloo.flix.api.lsp.{Index, InlayHint, InlayHintKind, Position, Range}
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Type, TypeConstructor}
+
+object InlayHintProvider {
+
+  def processInlayHints(uri: String, range: Range)(implicit index: Index, root: Root): List[InlayHint] = {
+    index.queryByRange(uri, range) match {
+      case Nil => Nil
+      case entities => entities.foldLeft(Nil: List[InlayHint]) {
+        case (acc, LocalVar(sym, tpe)) =>
+          getTypeHint(sym.loc, tpe).map(h => h :: acc).getOrElse(acc)
+        case (acc, _) =>
+          acc
+      }
+    }
+  }
+
+  private def getTypeHint(loc: SourceLocation, tpe: Type): Option[InlayHint] = {
+    val pos = Position(loc.endLine, loc.endCol)
+    getTypeText(tpe).map(label =>
+      InlayHint(pos, label, Some(InlayHintKind.Type), None, "", paddingRight = true)
```

Comment:
I guess we have to wait for actual VSCode support to see how this should be formatted.

Reply:
Yes, for now it's just my conjecture from what I have seen in other projects